### PR TITLE
added headers to auto generated files

### PR DESCRIPTION
**Pull request by k1ng440**
This pull request to resolve [#167](https://github.com/quick-lint/quick-lint-js/issues/167)

- [x] src/lex-keyword.cpp
- [x] src/translation-table-generated.cpp
- [x] src/quick-lint-js/translation-table-generated.h
- [x] translation .po files
- [x] doc/man/ files - headers are already there.

### CODE REVIEW

File: src/lex-keyword.cpp [hunk 1]
```diff
@@ -14,16 +17,16 @@
...
-    ('\\' == 92) && (']' == 93) && ('^' == 94) && ('_' == 95) &&               \
-    ('a' == 97) && ('b' == 98) && ('c' == 99) && ('d' == 100) &&               \
-    ('e' == 101) && ('f' == 102) && ('g' == 103) && ('h' == 104) &&            \
-    ('i' == 105) && ('j' == 106) && ('k' == 107) && ('l' == 108) &&            \
-    ('m' == 109) && ('n' == 110) && ('o' == 111) && ('p' == 112) &&            \
-    ('q' == 113) && ('r' == 114) && ('s' == 115) && ('t' == 116) &&            \
-    ('u' == 117) && ('v' == 118) && ('w' == 119) && ('x' == 120) &&            \
-    ('y' == 121) && ('z' == 122) && ('{' == 123) && ('|' == 124) &&            \
-    ('}' == 125) && ('~' == 126))
+    ('X' == 88) && ('Y' == 89) && ('Z' == 90) &&                               \
```
Comment (strager) [inline src/lex-keyword.cpp:20]: Hmm, these differences shouldn't happen. Are you using the wrong version of clang-format? See docs/CONTRIBUTING.md.

File: src/lex-keyword.cpp [hunk 2]
```diff
@@ -372,7 +375,7 @@ const struct keyword_entry *lexer_keyword::look_up(const char *str,
       if (o >= 0) {
         const char *s = o + stringpool;
 
-        if (*str == *s && !strncmp(str + 1, s + 1, len - 1) && s[len] == '\0')
+        if (*str == *s && !strncmp(str + 1, s + 1, len - 1) && s[len] == ' ')
```
Comment (strager) [inline src/lex-keyword.cpp:378]: Must fix: You introduced a syntax error.

File: src/lex-keyword.cpp [hunk 3]
```diff
@@ -1,3 +1,6 @@
+/* Code generated by tools/generate-lex-keyword. DO NOT EDIT. */
+/* source: src/lex-keyword.gperf */
```
Comment (strager) [inline src/lex-keyword.cpp:2]: Nit: Use `//` comments.

File: tools/compile-translations.go [hunk 4]
```diff
@@ -80,17 +80,19 @@
...
-import "bytes"
-import "fmt"
-import "io/ioutil"
-import "log"
-import "os"
-import "os/exec"
-import "path/filepath"
-import "sort"
-import "strings"
+import (
```
Comment (strager) [inline tools/compile-translations.go:83]: Undo unrelated changes like this one.

File: tools/generate-lex-keyword [hunk 5]
```diff
@@ -11,7 +11,9 @@ disable_line_directives() {
 }
 
 cd "$(dirname "${0}")/.."
-gperf src/lex-keyword.gperf | disable_line_directives >src/lex-keyword.cpp
+gperf src/lex-keyword.gperf | disable_line_directives > src/lex-keyword.cpp
+echo "/* Code generated by tools/generate-lex-keyword. DO NOT EDIT. */\n/* source: src/lex-keyword.gperf */\n\n$(cat src/lex-keyword.cpp)" > src/lex-keyword.cpp
```
Comment (strager) [inline tools/generate-lex-keyword:15]: Simpler:
```suggestion
{
  echo STUFF
  gperf src/lex-keyword.gperf | disable_line_directives
} >src/lex-keyword.cpp
```

File: src/lex-keyword.cpp (hunk 3 above)
Comment (strager) [inline src/lex-keyword.cpp:2]: Why does this file list a 'source', but other files don't?

**Comment by strager** [comment]
You said every generated file should mention 'do not edit', but the man pages don't and `src/global-variables-browser.cpp` doesn't either.

**Review by strager — CHANGES_REQUESTED** [review]

**Comment by strager** [comment]
src/translation-table-generated.cpp and src/quick-lint-js/translation-table-generated.h are sourced from po/*.po*.

po/messages.pot is sourced from a few .cpp files. See the script which generates it.

docs/man/* are sourced from docs/*.adoc.

src/global-variables-browser.cpp is sourced from many files within tools/browser-globals/ (really there's a separate repo with all the crap).

File: src/lex-keyword.cpp (hunk 1 above)
Comment (k1ng440) [reply src/lex-keyword.cpp:20]: yea i was using clang-format 10

File: src/lex-keyword.cpp (hunk 3 above)
Comment (k1ng440) [reply src/lex-keyword.cpp:2]: Because I do know the source files for most of them. i will try to add rest of them

Comment (k1ng440) [reply src/lex-keyword.cpp]: done

File: src/lex-keyword.cpp (hunk 2 above)
Comment (k1ng440) [reply src/lex-keyword.cpp]: I believe it was caused by clang-format. downgrading to v9 solved the issue

File: tools/compile-translations.go (hunk 4 above)
Comment (k1ng440) [reply tools/compile-translations.go:83]: golangci-lint makes the multiple import to a group of imports

Comment (k1ng440) [reply tools/compile-translations.go:83]: even the `staticcheck` does it which is the default formatter for golang

**Comment by k1ng440** [comment]
@strager there is no bash file for generating global-variables-browser.cpp. should I create one? see `tools/browser-globals/README.md`

**Comment by strager** [comment]
> @strager there is no bash file for generating global-variables-browser.cpp. should I create one?

No.

**Comment by k1ng440** [comment]
I have read and hereby agree to quick-lint-js' Contributor License Agreement (CLA-v1.md).

File: src/quick-lint-js/translation-table-generated.h [hunk 6]
```diff
@@ -1,11 +1,12 @@
+// Code generated by tools/compile-translations.go. DO NOT EDIT.
+// source po/*.po
```
Comment (strager) [inline src/quick-lint-js/translation-table-generated.h:2]: Shouldn't this have a colon:
```suggestion
// source: po/*.po
```

File: tools/update-translator-sources [hunk 7]
```diff
@@ -34,9 +34,20 @@ update_template_file() {
...
   msgmerge --backup=none --update -- "${pot_file}" "${pot_temp_file}"
   sed -i -e '/^.POT-Creation-Date:/d' "${pot_file}"
 
+	if [ $(cat $pot_file | grep -c "Code generated by") -eq 0 ]; then
+		sed -i '1s;^;# Code generated by tools/update-translator-sources. DO NOT EDIT.\n;' "$pot_file"
+		sed -i '2s;^;# source: benchmark/benchmark-translation.cpp\n;' "$pot_file"
+		sed -i '3s;^;# source: src/diagnostic-formatter.cpp\n;' "$pot_file"
+		sed -i '4s;^;# source: src/quick-lint-js/diagnostic-types.h\n;' "$pot_file"
+		sed -i '5s;^;# source: test/test-diagnostic-formatter.cpp\n;' "$pot_file"
+		sed -i '6s;^;# source: test/test-vim-qflist-json-diag-reporter.cpp\n\n;' "$pot_file"
```
Comment (strager) [inline tools/update-translator-sources:48]: Avoid duplicating the list of files from line 29.

I think we can just write `source: (various)` here. No need to be thorough.

File: tools/update-translator-sources [hunk 8]
```diff
@@ -34,9 +34,20 @@ update_template_file() {
   sed -i -e 's/charset=CHARSET/charset=UTF-8/g' "${pot_temp_file}"
 
   mkdir -p po
+
+
   msgmerge --backup=none --update -- "${pot_file}" "${pot_temp_file}"
   sed -i -e '/^.POT-Creation-Date:/d' "${pot_file}"
 
+	if [ $(cat $pot_file | grep -c "Code generated by") -eq 0 ]; then
```
Comment (strager) [inline tools/update-translator-sources:42]: Nit:
```suggestion
	if [ $(grep -c "Code generated by" "$pot_file") -eq 0 ]; then
```

File: tools/update-translator-sources [hunk 9]
```diff
@@ -34,9 +34,20 @@ update_template_file() {
   sed -i -e 's/charset=CHARSET/charset=UTF-8/g' "${pot_temp_file}"
 
   mkdir -p po
+
+
```
Comment (strager) [inline tools/update-translator-sources:38]: Nit: undo

**Review by strager — APPROVED** [review]

File: tools/update-translator-sources (hunk 8 above)
Comment (strager) [inline tools/update-translator-sources:42]: Another way to write this:
```suggestion
	if ! grep -q "Code generated by" "${pot_file}"; then
```

File: src/quick-lint-js/translation-table-generated.h [hunk 10]
```diff
@@ -1,12 +1,16 @@
...
 // Copyright (C) 2020  Matthew "strager" Glazar
 // See end of file for extended copyright information.
 
+
 #ifndef QUICK_LINT_JS_TRANSLATION_TABLE_GENERATED_H
 #define QUICK_LINT_JS_TRANSLATION_TABLE_GENERATED_H
 
+
+
+
```
Comment (strager) [inline src/quick-lint-js/translation-table-generated.h]: Must fix: These changes reappeared after I ran `./tools/update-translator-sources`. Fix the code generator.

**Review by strager — CHANGES_REQUESTED** [review]

File: src/quick-lint-js/translation-table-generated.h (hunk 10 above)
Comment (k1ng440) [reply src/quick-lint-js/translation-table-generated.h]: ok. I have fixed it. 

the blank lines was introduced  in `tools/compile-translations.go`

**Comment by strager** [comment]
Landed as commit 49217927c553b4cf199faf0cfdae58ec9a324e50.
